source/YTEPs/YTEP-0037.rst

```diff
@@ -0,0 +1,133 @@
+YTEP-0037: Code styling
+=======================
+
+Abstract
+--------
+
+Created: May 18, 2020
+Author: Clément Robert
+
+This YTEP proposes the enforcement of code styling guidelines with auto-formatting tools.
+
+
+Status
+------
+
+Proposed
+
+Project Management Links
+------------------------
+
+The follow PR corpus demonstrates the feasibility of the proposal
+
+  * sorting imports with `isort <https://github.com/timothycrosley/isort>`_ (`#2592 <https://github.com/yt-project/yt/pull/2592>`_)
+  * code formatting with `black <https://github.com/psf/black>`_ (`#2596 <https://github.com/yt-project/yt/pull/2596>`_)
+  * redirect internal imports from the wrapper module ``yt.units`` to ``unyt`` (`#2597 <https://github.com/yt-project/yt/pull/2597>`_)
+  * add a ``pyproject.toml`` file (`#2598 <https://github.com/yt-project/yt/pull/2598>`_)
+  * add a precommit hook configuration file (`#2600 <https://github.com/yt-project/yt/pull/2600>`_)
+
+Detailed Description
+--------------------
+
+Code styling guidelines are already presented in the `project's documentation
+<https://yt-project.org/docs/dev/developing/developing.html#coding-style-guide>`_,
+though enforcing them is not explicitly made part of the reviewing process.
+We already use ``flake8``, and integrate it to our CI to catch a subset of infractions
+to PEP8. 
+
+``black`` automatically enforces a superset of those rules, and offers very little
+configuration option by design. Only the target line length can be changed.
+This makes it the only point requiring discussion if this YTEP is approved.
+
+**maximal line length**
+
+The guidelines states that
+
+    Line widths should not be more than 80 characters
+
+Despite this being respected in most of the code base, there remain a large amount of
+outliers, that would be time-consuming to go through by hand. Taking the example of the
+yt-4.0 branch at the time of writing, there are 2676 lines exceeding 80 characters (~2%
+of the whole code base), or, visually
+
+.. image:: ../images/lw-ytep-35.png
+
+Note that in some cases, ``black`` will leave long lines as is. For instance, it won't
+split a 200-chars long string. 
+
+There is a range of possible values we might give preference to. Python's standard
+library caps linelenght at 79, pandas does so at 80. By default, ``black`` will target
+88, as its authors claim it reduces the total number of lines by some 10% (as compared
+to enforcing 80).
+
+In first drafting the PR linked above, I chose a line-lenght of 100, so as to minimize
```

Comment:
I think we should go with something in line with either stdlib or pandas.  I'd be fine with 79 or 80.

Reply:
Personally I would align with black’s default, which is also compatible with Raymond Hettinger’ style of «90-ish». It makes for much shorter files (according to black).

Reply:
That'd work for me.

Reply:
I'm ok with 88.

Reply:
btw I double checked and actually pandas uses 88, not 80 as I previously reported.